Rebase the patch to latest version.

By Dongxiao Xu <dongxiao.xu@intel.com>

Upstream-Status: Pending

diff -ruN gsm-orig/src/gsmd/gsmd.c gsm/src/gsmd/gsmd.c
--- gsm-orig/src/gsmd/gsmd.c	2010-11-25 10:14:22.000000000 +0800
+++ gsm/src/gsmd/gsmd.c	2010-11-25 10:14:52.000000000 +0800
@@ -196,14 +196,11 @@
 
 	sms_cb_init(gsmd);
 
-	if (gsmd->vendorpl && gsmd->vendorpl->initsettings){
+	if (gsmd->vendorpl && gsmd->vendorpl->initsettings)
 		rc |= gsmd->vendorpl->initsettings(gsmd);
-		if (gsmd->machinepl && gsmd->machinepl->initsettings)
-			rc |= gsmd->machinepl->initsettings(gsmd);
-		return rc;
-	}	
-	else
-		return rc;
+	if (gsmd->machinepl && gsmd->machinepl->initsettings)
+		rc |= gsmd->machinepl->initsettings(gsmd);
+	return rc;
 }
 
 static int firstcmd_response = 0;
